src/node.js

```diff
@@ -22,6 +22,44 @@
 
     EventEmitter.call(process);
 
+    // Default process warning handler..
+    process.on('warning', (warning) => {
+      // By default, print the short warning message if --no-warnings is not
+      // set. If --trace-warnings is set, then the stack trace for the warning
+      // is shown. This takes backwards compatibility with --trace-deprecation
+      // into consideration also.
+      if (process.noProcessWarnings) return;
+      const trace = process.traceProcessWarnings ||
+                    (warning.name === 'DeprecationWarning' &&
+                     process.traceDeprecation);
+      if (trace)
+        console.error(warning.stack);
+      else
+        console.error(`${warning.name}: ${warning.message}`);
+    });
+
+    // process.emitWarning(error)
+    // process.emitWarning(str)
+    // process.emitWarning(options)
+    var Warning;
+    process.emitWarning = function(warning) {
+      if (!Warning)
+        Warning = NativeModule.require('internal/warning');
+      if (typeof warning === 'string') {
+        warning = Warning('Warning', warning, process.emitWarning);
+      } else if (!(warning instanceof Error) && typeof warning === 'object') {
+        if (warning.message) {
+          warning = Warning(`${warning.name || 'Warning'}`,
+                            `${warning.message}`,
+                            process.emitWarning);
+        }
+      }
+      if (!(warning instanceof Error))
+        throw new TypeError('\'warning\' must be an Error object, string, or' +
```

Comment:
I noticed this too, the eslint rules need to be relaxed for this file to allow template strings, right?